lib/about_page/configuration.rb

```diff
@@ -63,21 +64,35 @@ def set_headers! response
     end
 
     def to_h
-      self.hash.to_h.inject({}) { |h,v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
+      self.hash.to_h.inject({}) { |h, v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
     end
 
     def health_report
-      self.nodes.collect do |key, profile| 
-        if profile.class.validators.length > 0 
-          health = profile.valid? ? 'ok' : 'error'
-          errors = profile.errors.map {|a,m| "#{a} #{m}"}
-          { 'component' => key.to_s, 'status' => health, 'errors' => errors }
-        else
-          nil
+      self.nodes.collect do |key, profile|
+        next unless profile.class.validators.length.positive?
+
+        begin
+          health = Timeout.timeout(55) { profile.valid? ? 'ok' : 'error' }
+          errors = error_message(profile)
+        rescue Timeout::Error
+          health = 'error'
+          # ActiveModel::Errors.add expects the first argument to be the name
+          # of the method that generated the error. We do not have access to the
+          # name of the validator that timed out in this context, so define a
+          # dummy `timeout` method to reference when adding the error to the model.
+          profile.class.define_method(:timeout) {}
+          profile.errors.add(:timeout, message: ': component check has timed out.')
+          errors = error_message(profile)
         end
+
+        { 'component' => key.to_s, 'status' => health, 'errors' => errors }
       end.compact
     end
 
+    def error_message(profile)
+      profile.errors.messages.map { |a, m| "#{a} #{m.join(', ')}" }
```

Comment:
`ActiveModel::Errors` was reworked in Rails 6.1 so that it is no longer a hash like object so mapping directly on it is no longer supported. We have to access the messages which is a hash of arrays e.x. `{ attr 1: [error 1, error 2, etc], attr 2: [] }` and we map over that.